ui/src/app/pages/workspace/workspace-wrapper.tsx

```diff
@@ -67,7 +72,7 @@ export const WorkspaceWrapper = fp.flow(
       nextWorkspaceWarmupStore.next(undefined);
       if (nextWs && nextWs.namespace === ns && nextWs.id === wsid) {
         currentWorkspaceStore.next(nextWs);
-        return;
+        updateStores(ns);
```

Comment:
nit: one small refactoring we can do here is to just set `nextWs` with a call to `getWorkspace()` if `nextWs` is null and then run the `updateStores` call on that

Reply:
unfortunately, this is set up like this because you can't call async/await directly in a useEffect.